name: test

on:
  push:
    branches:
      - main
  pull_request:

concurrency:
  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
  cancel-in-progress: true

env:
  LOG_LEVEL: debug

permissions:
  contents: read

jobs:
  lint:
    runs-on: ubuntu-latest
    timeout-minutes: 10
    steps:
      - name: ⚙️ Setup
        uses: containerbase/internal-tools/setup@ef820394d9e9464ea11590dea1487f5f9cef0603 # v3.10.60
        with:
          save-cache: true

      - run: pnpm prettier

  preset:
    runs-on: ubuntu-latest
    timeout-minutes: 10
    container:
      image: ghcr.io/renovatebot/renovate:41.43.0@sha256:aba4892ec62ab5f972304cf24d7ddc517ef80288ab23d3e7868546eee3a3a10c
      # github hosted runners are running as `1001:127` (ubuntu:docker)
      options: -u 1001:0 --group-add 1000 --group-add 12021 --group-add 127

    strategy:
      matrix:
        file:
          - default
          - renovate-config
          - merge-queue

    steps:
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
        with:
          show-progress: false

      - name: testing ${{ matrix.file }}
        run: renovate-config-validator ${{ matrix.file }}.json

  preset-success:
    runs-on: ubuntu-latest
    needs:
      - lint
      - preset
    timeout-minutes: 1
    if: always()
    steps:
      - name: Fail for failed or cancelled presets
        if: |
          needs.test.result == 'failure' ||
          needs.test.result == 'cancelled'
        run: exit 1
      - name: Fail for failed or cancelled lint
        if: |
          needs.lint.result == 'failure' ||
          needs.lint.result == 'cancelled'
        run: exit 1
